### ISSUE TITLE
Segmentation ID's

### ISSUE THREAD

**Issue by rachthree** [microsoft/AirSim-NeurIPS2019-Drone-Racing#92]
Hello,

Is there a way to get segmentation ID's using the API when an image is taken (for training purposes, not during the race)? Right now, I get RGB when using `simGetImages([airsim.ImageRequest(camera_name, airsim.ImageType.Segmentation, pixels_as_float=False, compress=False)])` according to the mapping (https://github.com/microsoft/AirSim/blob/master/docs/seg_rgbs.txt, which I think is off... something I know is segmentation ID 0 shows as `[130, 219, 128]` instead of `[55, 181, 57]`). I can think of a workaround, but if there is an API call to know the segmentation ID off the bat when an image is taken (especially if it's an alpha channel), that would be very helpful.

Thanks!

Edit: To clarify, I'm trying to have an image annotated with masks of segmentation ID's either as an alpha channel or RGB with all 3 channels having the same number to represent the ID. Currently the segmentation images are colored by the color palette, not by the ID itself. For example, if a pixel is associated with ID 0, I'd like to see `[0, 0, 0]` in the segmentation image rather than `[55, 181, 57]` or 0 in an alpha channel of the RGB `Scene` image. Is this readily possible with AirSim?

**Comment by madratman** [comment]
use this as your ~/Documents/AirSim/settings.json

```json
{
  "ClockSpeed": 1,
  "SeeDocsAt": "https://github.com/Microsoft/AirSim/blob/master/docs/settings.md",
  "SettingsVersion": 1.2,
  "SimMode": "Multirotor",
  "Vehicles": {
    "drone_1": {
      "Cameras": {
        "fpv_cam": {
          "CaptureSettings": [
            {
              "FOV_Degrees": 90,
              "Height": 240,
              "ImageType": 0,
              "Width": 320
            },
            {
              "Height": 240,
              "ImageType": 5,
              "Width": 320
            }
          ],
          "Pitch": 0.0,
          "Roll": 0.0,
          "X": 0.25,
          "Y": 0.0,
          "Yaw": 0.0,
          "Z": 0.0
        }
      },
      "Pitch": 0.0,
      "Roll": 0.0,
      "VehicleType": "SimpleFlight",
      "X": 0.0,
      "Y": 0.0,
      "Yaw": 0.0,
      "Z": 0.0
    },
    "drone_2": {
      "Pitch": 0.0,
      "Roll": 0.0,
      "VehicleType": "SimpleFlight",
      "X": 0.0,
      "Y": 0.0,
      "Yaw": 0.0,
      "Z": 0.0
    }
  }
}
```

And you get the rgb and segmentation images as:


```python
requests = [airsim.ImageRequest("fpv_cam", airsim.ImageType.Scene, False, False), airsim.ImageRequest("fpv_cam", airsim.ImageType.Segmentation, False, False)]
responses = self.airsim_client_images.simGetImages(request)
# responses[0] will be rgb
# responses[1] will be segmentation 
```

re: segmentation ids - that file is old. 
You can do set the segmentation of ID of all object except gates to be the same number by using [simSetSegmentationObjectID](https://microsoft.github.io/AirSim-NeurIPS2019-Drone-Racing/api.html#airsimneurips.client.MultirotorClient.simSetSegmentationObjectID), then you can do a numpy.unique on the segmentation image to figure out the gate rgb value, as at the end of the day, you need a binary segmentation mask

**Comment by rachthree** [comment]
Thank you for the suggestion! That is very similar to what I had to do, so I’m glad to hear I was going in the right direction. I ended up pointing the computer vision camera at the ground, changed the ground segmentation ID for all ID’s, and mapped the resulting segmentation RGB to the ID.  It would be nice for AirSim to provide such masks already as this would greatly help utilize it for perception with deep learning methods. I will probably create an issue there after the competition.
